classes/output/mobile.php

```diff
@@ -0,0 +1,234 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Mobile output class for mod_questionnaire.
+ *
+ * @copyright 2018 Igor Sazonov <sovletig@gmail.com>
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace mod_questionnaire\output;
+
+defined('MOODLE_INTERNAL') || die();
+
+class mobile {
+
+    /**
+     * Returns the initial page when viewing the activity for the mobile app.
+     *
+     * @param  array $args Arguments from tool_mobile_get_content WS
+     * @return array HTML, javascript and other data
+     */
+    public static function mobile_view_activity($args) {
+        global $OUTPUT, $USER, $CFG, $DB;
+
+        require_once($CFG->dirroot.'/mod/questionnaire/questionnaire.class.php');
+        $args = (object) $args;
+        $cmid = $args->cmid;
+        $pagenum = (isset($args->pagenum) && !empty($args->pagenum)) ? intval($args->pagenum) : 1;
+        $prevpage = 0;
+        // Capabilities check.
+        $cm = get_coursemodule_from_id('questionnaire', $cmid);
+        $context = \context_module::instance($cmid);
+        self::require_capability($cm, $context, 'mod/questionnaire:view');
+        // Set some variables we are going to be using.
+        $questionnaire = get_questionnaire_data($cmid, $USER->id);
+        if (isset($questionnaire['questions'][$pagenum-1]) && !empty($questionnaire['questions'][$pagenum-1])) {
+            $prevpage = $pagenum-1;
+        }
+        //$jsfilepath = $CFG->wwwroot . '/mod/questionnaire/javascript/mobile.js';
```

Comment:
Any reason why this is commented out

Reply:
I can actually remove that, it's just a debug thanks